rust_src/src/buffers.rs

```diff
@@ -814,26 +814,28 @@ pub fn overlay_properties(overlay: LispOverlayRef) -> LispObject {
 
 #[no_mangle]
 pub unsafe extern "C" fn validate_region(b: *mut LispObject, e: *mut LispObject) {
-    let start = *b;
-    let stop = *e;
+    let (begin, end) = validate_region_rust(*b, *e);
+    *b = begin.into();
+    *e = end.into();
+}
 
-    let mut beg = start.as_fixnum_coerce_marker_or_error();
+pub fn validate_region_rust(start: LispObject, stop: LispObject) -> (isize, isize) {
```

Comment:
Can the inputs be made to be more specific than `LispObject`? Same for all the callers.

Reply:
I did not think we had a match for "fixnum or coerced marker".

Reply:
I need to research this more. There is `CHECK_FIXNUM_COERCE_MARKER` and `CHECK_NUMBER_COERCE_MARKER` in the original Emacs source. We have a representation for the NUMBER version called `LispNumber`. We do not have a similar type for the FIXNUM variant currently. There are matching `Qinteger_or_marker_p` and `Qnumber_or_marker_p` as well.

I need to look and see what the Emacs source was expecting and then evaluate if a more explicit type is possible.

Reply:
I believe the fixnum variant is new, now that upstream has introduced arbitrary sized integers.